tests/utils/state-locks-filter

```diff
@@ -0,0 +1,402 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import re
+import sys
+
+"""
+LockTrace: Represents a trace in the log file.
+It holds the lines for the trace and allows matching against a part of the trace.
+"""
+
+
+class LockTrace:
+    def __init__(self, lines: list[str]):
+        self.lines = lines
+
+    def get_trace_lines(self) -> list[str]:
+        return self.lines
+
+    def match(self, part: str) -> bool:
+        for line in self.lines:
+            if part in line:
+                return True
+
+        return False
+
+    def __str__(self) -> str:
+        return "".join(self.lines)
+
+    def __hash__(self):
+        return hash(str(self))
+
+    def __eq__(self, other) -> bool:
+        if not isinstance(other, LockTrace):
+            # don't attempt to compare against unrelated types
+            return NotImplemented
+
+        return hash(str(self)) == hash(str(other))
+
+
+"""
+Lock: Represents a lock in the log file, including its header and times (held and wait times).
```

Comment:
wouldn't LockOp and "represents a lock operation" be clearer?  same possibly for Lock in the name of other classes here